flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackend.java

```diff
@@ -104,6 +104,25 @@ default String getName() {
     <K> CheckpointableKeyedStateBackend<K> createKeyedStateBackend(
             KeyedStateBackendParameters<K> parameters) throws Exception;
 
+    /**
+     * Creates a new {@link AsyncKeyedStateBackend} which supports to access <b>keyed state</b>
+     * asynchronously.
+     *
+     * <p><i>Keyed State</i> is state where each value is bound to a key.
+     *
+     * @param parameters The arguments bundle for creating {@link AsyncKeyedStateBackend}.
+     * @param <K> The type of the keys by which the state is organized.
+     * @return The Async Keyed State Backend for the given job, operator.
+     * @throws Exception This method may forward all exceptions that occur while instantiating the
+     *     backend.
+     */
+    @Experimental
+    default <K> AsyncKeyedStateBackend createAsyncKeyedStateBackend(
+            KeyedStateBackendParameters<K> parameters) throws Exception {
+        throw new UnsupportedOperationException(
+                "Don't support createAsyncKeyedStateBackend by default");
+    }
+
```

Comment:
Should we add another method `isSupportAsyncKeyedStateBackend` within `StateBackend`?

Reply:
Make sense. I just added `supportsAsyncKeyedStateBackend`.
The naming is consistent with `supportsNoClaimRestoreMode` and `supportsSavepointFormat`.